src/lib.rs

```diff
@@ -69,31 +68,40 @@ impl FxHasher {
 }
 
 impl Hasher for FxHasher {
+    #[cfg(target_pointer_width = "32")]
+    #[inline]
+    fn write(&mut self, mut bytes: &[u8]) {
+        while bytes.len() >= 4 {
+            self.add_to_hash(NativeEndian::read_u32(bytes) as usize);
+            bytes = &bytes[4..];
+        }
+        if bytes.len() >= 2 {
+            self.add_to_hash(NativeEndian::read_u16(bytes) as usize);
+            bytes = &bytes[2..];
+        }
+        if !bytes.is_empty() {
+            self.add_to_hash(bytes[0] as usize);
+        }
+    }
+
+    #[cfg(target_pointer_width = "64")]
     #[inline]
     fn write(&mut self, mut bytes: &[u8]) {
-        #[cfg(target_pointer_width = "32")]
-        let read_usize = |bytes| NativeEndian::read_u32(bytes);
-        #[cfg(target_pointer_width = "64")]
-        let read_usize = |bytes| NativeEndian::read_u64(bytes);
-
-        let mut hash = FxHasher { hash: self.hash };
```

Comment:
This variable should be kept, so we don't load/store from memory multiple times.
Also you should verify that the assembly and LLVM IR output isn't changed.

Reply:
Oh I see thanks, I've added it back with a comment.